test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/MaxInputBufferLengthTests.cs

```diff
@@ -0,0 +1,173 @@
+﻿using System;
+using System.Net;
+using System.Net.Sockets;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.FunctionalTests
+{
+    public class MaxInputBufferLengthTests
+    {
+        [Theory]
+        [InlineData(16 * 1024, true, true)]
+        [InlineData(16 * 1024, false, true)]
+        [InlineData(1024 * 1024, true, true)]
+        [InlineData(1024 * 1024, false, true)]
+        [InlineData(5 * 1024 * 1024, true, true)]
+        [InlineData(5 * 1024 * 1024, false, true)]
+        [InlineData(10 * 1024 * 1024, true, false)]
+        [InlineData(10 * 1024 * 1024, false, false)]
+        [InlineData(Int32.MaxValue, true, false)]
+        [InlineData(Int32.MaxValue, false, false)]
+        [InlineData(-1, true, false)]
+        [InlineData(-1, false, false)]
```

Comment:
It's not allowed.  Valid values are currently `>1` or `-1` (which will soon be replaced by `null`).  I will add a test for a tiny buffer like `1`, which should pause the connection after each packet.